lib/rules/no-legacy-test-waiters.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const LEGACY_TEST_WAITER_FUNCTIONS = ['registerWaiter', 'unregisterWaiter'];
+const ERROR_MESSAGE =
+  'The use of the legacy test waiters API is not allowed. Please use the new ember-test-waiters (https://github.com/emberjs/ember-test-waiters) APIs instead.';
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'disallow the use of the legacy test waiter APIs.',
+      category: 'Testing',
+      recommended: false,
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-legacy-test-waiters.md',
+    },
+    fixable: null,
+    schema: [],
+  },
+
+  ERROR_MESSAGE,
+
+  create(context) {
+    let testWaitersImportFound = false;
+
+    return {
+      ImportDeclaration(node) {
+        if (node.source.value !== '@ember/test') {
+          return;
+        }
+
+        node.specifiers.forEach(specifier => {
+          testWaitersImportFound = LEGACY_TEST_WAITER_FUNCTIONS.includes(specifier.imported.name);
```

Comment:
Instead of just checking that the import is found, we should also store what name the legacy test wait functions are imported under.

Initially we start out with an empty object:
```
const legacyTestWaiterImportMappings = {};
```

If we see this import statement:
```
import { registerWaiter } from '@ember/test';
```
then we store it with `legacyTestWaiterImportMappings.registerWaiter = 'registerWaiter';`

Or if we see:
```
import { registerWaiter as someOtherName } from '@ember/test';
```

then we store it with `legacyTestWaiterImportMappings.registerWaiter = 'someOtherName';`

That way, we know what function name to look for later.

Reply:
Ya we have a utility that gathers import aliases for our internal eslint plugins. That may be useful here. What do you think?

Reply:
That would be great. I would LOVE it if you helped us add/improve util functions (likely in `utils/import.js`) for checking imports. A lot of our rules are missing import checks today or not checking imports correctly.

Reply:
Done. Added tests for it too!